master/internal/rm/agentrm/agent.go

```diff
@@ -129,10 +132,40 @@ func (a *agent) receive(ctx *actor.Context, msg interface{}) error {
 		a.slots, _ = ctx.ActorOf("slots", &slots{})
 	case model.AgentSummary:
 		ctx.Respond(a.summarize(ctx))
-	case ws.WebSocketConnected:
-		check.Panic(check.True(a.socket == nil, "websocket already connected"))
-		socket, ok := msg.Accept(ctx, aproto.MasterMessage{}, true)
-		check.Panic(check.True(ok, "failed to accept websocket connection"))
+	case actorapi.WebSocketRequest:
+		if a.socket != nil {
+			panic("websocket already connected")
+		}
+
+		conn, err := ws.UpgradeEchoConnection(msg.Ctx)
+		if err != nil {
+			ctx.Log().WithError(err).Panic("Error upgrading connection to WebSocket")
+		}
+
+		wsName := "master-agent-ws-" + ctx.Self().Address().String()
+		socket, err := ws.Wrap[*aproto.MasterMessage, aproto.AgentMessage](wsName, conn)
+		if err != nil {
+			ctx.Log().WithError(err).Panic("failed to accept websocket connection")
+		}
+
+		// spin up goroutine that sends messages to self
+		go func() {
+		InboxLoop:
```

Comment:
`break InboxLoop` will break the loop that follows the label rather than the `switch` enclosing the `break` statement. `continue` would just carry on to the next iteration of the loop, which would be extremely bad in the case that `socket.Inbox` is closed (since this would busy loop until `socket.Done` was closed).

Maybe I should defer the call to `ctx.Tell` and `return` rather than using a label, which might be more clear.